app/logic/serviceLearningCourses.py

```diff
@@ -77,28 +77,74 @@ def saveCourseParticipantsToDatabase(cpPreview: Dict[str, Dict[str, Dict[str, Li
                 print(f"Unable to save course {course}. {courseInfo['errorMsg']}")
                 continue
 
-            courseObj: Course = Course.get_or_create(
-                 courseAbbreviation = course,
-                 term = termObj, 
-                 defaults = {"CourseName" : "",
-                             "sectionDesignation" : "",
-                             "courseCredit" : "1",
-                             "term" : termObj,
-                             "status" : 4,
-                             "createdBy" : g.current_user,
-                             "serviceLearningDesignatedSections" : "",
-                             "previouslyApprovedDescription" : "" })[0]
-
-            for userDict in courseInfo['students']:
-                if userDict['errorMsg']:
+            currentTermObj = Term.get(Term.id == courseTerm)
+
+            existingCourses = list((Course.select()
+                                            .join(Term, on =(Course.term == Term.id))
+                                            .where((Course.courseAbbreviation == course) & (Term.termOrder <= currentTermObj.termOrder))
+                                            .order_by(Course.term.desc())
+                                            .limit(1)))
+
+            if not existingCourses :
+                
+                courseObj: Course = Course.create(courseName = "",
+                                sectionDesignation = "",
+                                courseAbbreviation = course,
+                                courseCredit = "1",
+                                term = courseTerm,
+                                status = CourseStatus.IN_PROGRESS,
+                                createdBy = g.current_user,
+                                serviceLearningDesignatedSections = "",
+                                previouslyApprovedDescription = "")
+                CourseQuestion.create(course = courseObj.id,
+                                questionContent= " ",
+                                questionNumber=" ")
+                CourseInstructor.create(course = courseObj.id,
+                                    user = 'ramsayb2')
+            else:
+                previousMatchedCourse = existingCourses[0]
+
+                currentTerm = Term.get(Term.id == previousMatchedCourse.term)
```

Comment:
`currentTerm` isn't a great name, because it's not the current term in the application, nor is it necessarily the same as the `currentTermObj` term